bats/tests/helpers/defaults.bash

```diff
@@ -201,6 +195,14 @@ using_dev_mode() {
 }
 
 ########################################################################
+# Kubernetes versions
+
+# The main Kubernetes version to test.
+: "${RD_KUBERNETES_VERSION:=1.23.6}"
```

Comment:
Something feels odd about having `RD_KUBERNETES_VERSION` and `RD_K3S_VERSIONS`, but I'm not sure what I would like to change.

Logically these are the same settings, with `RD_K3S_VERSIONS` even defaulting to `RD_KUBERNETES_VERSION`. But calling the former `RD_KUBERNETES_VERSIONS` means the names are almost identical save for the final `S` which feels confusing too.

So maybe we'll leave everything as is for now, but it doesn't really feel right to me.

Reply:
So I'm just brainstorming here, and we don't have to do it in this PR, but how about renaming

1. `RD_KUBERNETES_VERSION` → `RD_K3S_VERSION`
2. `RD_K3S_VERSIONS` → `RD_K3S_VERSION_LIST`

Reply:
I think I'll be happier with `RD_KUBERNETES_VERSION`, `RD_KUBERNETES_VERSION_LIST`; what do you think?  (And yeah, not for this PR.)

Reply:
Yes, `RD_KUBERNETES_VERSION_LIST` is probably better. The only reason I considered `RD_K3S_VERSION_LIST` is because it is shorter. 😄 

Note that we should then also rename `RD_K3S_MIN` and `RD_K3S_MAX`.

But where do we stop? Do we also rename `foreach_k3s_version` and `get_k3s_versions`?

I think I would like to use consistent naming for all, but now `kubernetes` is looking quite long again. There is a reason it is often written as `k8s`. I guess I'm still fine with using `KUBERNETES` everywhere; it's not like we have to type these names a lot.